server/controllers/finance/purchases.js

```diff
@@ -346,7 +389,39 @@ function purchaseStatus(req, res, next) {
       }
       return db.exec(query, [purchaseUuid]);
     })
-    .then(() => res.status(200).send(status))
+    .then(() => {
+      /**
+        * Get all the inventories of a purchase order finally to obtain the average waiting time between the order and the delivery
+      */
+      const getInventory = `
+        SELECT BUID(purchase_item.inventory_uuid) AS inventory_uuid, inventory.delay
+        FROM purchase_item
+        JOIN inventory ON inventory.uuid = purchase_item.inventory_uuid
+        WHERE purchase_item.purchase_uuid = ?
+      `;
+
+      return db.exec(getInventory, [purchaseUuid]);
+    })
+    .then((rows) => {
+      const transaction = db.transaction();
+
+      rows.forEach((row) => {
+        /**
+          * Normally the delay agreement time between the order and the delivery is calculated 
+          * by finding the average duration of agreement of orders of last six months for a product 
+          * this calculation is very expensive in terms of memory, which is the reason why we keep 
+          * this information in the inventory table for article shovel
+        */
+
+        let delay = row.delay ? ((status.delay + row.delay) / 2) : status.delay;
```

Comment:
The same idea can be applied as above.  Just add a `total_delay` column to the `purchase_total` table.